infra/modules/aurora-db/main.tf

```diff
@@ -0,0 +1,201 @@
+
+resource "aws_backup_plan" "postgresql" {
+  name = "${var.service_name}_backup_plan"
+
+  rule {
+    rule_name         = "${var.service_name}_backup_rule"
+    target_vault_name = "${var.service_name}-vault"
+    schedule          = "cron(0 12 ? * SUN *)"
+  }
+}
+
+# KMS Key for the vault
+# This key was created by AWS by default alongside the vault
+data "aws_kms_key" "postgresql" {
+  key_id = "alias/aws/backup"
+}
+# create backup vault
+resource "aws_backup_vault" "postgresql" {
+  name        = "${var.service_name}-vault"
+  kms_key_arn = data.aws_kms_key.postgresql.arn
+}
+
+# create IAM role
+resource "aws_iam_role" "postgresql_backup" {
+  name_prefix        = "aurora-backup-"
+  assume_role_policy = data.aws_iam_policy_document.postgresql_backup.json
+}
+
+resource "aws_iam_role_policy_attachment" "postgresql_backup" {
+  role       = aws_iam_role.postgresql_backup.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSBackupServiceRolePolicyForBackup"
+}
+
+data "aws_iam_policy_document" "postgresql_backup" {
+  statement {
+    actions = [
+      "sts:AssumeRole",
+    ]
+
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["backup.amazonaws.com"]
+    }
+  }
+}
+# backup selection
+resource "aws_backup_selection" "postgresql_backup" {
+  iam_role_arn = aws_iam_role.postgresql_backup.arn
+  name         = "${var.service_name}-backup"
+  plan_id      = aws_backup_plan.postgresql.id
+
+  resources = [
+    aws_rds_cluster.postgresql.arn
+  ]
+}
+
+################################################################################
+# IAM role for enhanced monitoring
+################################################################################
+
+resource "aws_iam_role" "rds_enhanced_monitoring" {
+  name_prefix        = "aurora-enhanced-monitoring-"
+  assume_role_policy = data.aws_iam_policy_document.rds_enhanced_monitoring.json
+}
+
+resource "aws_iam_role_policy_attachment" "rds_enhanced_monitoring" {
+  role       = aws_iam_role.rds_enhanced_monitoring.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonRDSEnhancedMonitoringRole"
+}
+
+data "aws_iam_policy_document" "rds_enhanced_monitoring" {
+  statement {
+    actions = [
+      "sts:AssumeRole",
+    ]
+
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["monitoring.rds.amazonaws.com"]
+    }
+  }
+}
+
+################################################################################
+# Parameters for Query Logging
+################################################################################
+
+resource "aws_rds_cluster_parameter_group" "rds_query_logging" {
+  name        = var.service_name
+  family      = "aurora-postgresql13"
+  description = "Default cluster parameter group"
+
+  parameter {
+    name  = "log_statement"
+    value = "ddl" 
+  }
+
+  parameter {
+    name  = "log_min_duration_statement"
+    value = "1"
```

Comment:
[this doc](https://www.postgresql.org/docs/current/runtime-config-logging.html) makes it seem like the units is in milliseconds, so this is saying we should log query times that exceed 1 ms, which seems excessive. what about "100 ms"? (also could be useful to add a comment explaining the units)